themes/default/content/docs/intro/concepts/inputs-outputs.md

```diff
@@ -651,6 +651,327 @@ variables:
+            "Hello": {
+                "Type": "Task",
+                "Resource": helloFunction.arn,
+                "Next": "World",
+            },
+            "World": {
+                "Type": "Task",
+                "Resource": worldFunction.arn,
+                "End": true,
+            },
+        },
+    })
+});
+```
+
+### Example Parsing JSON Outputs to Objects
+
+The following is an example of Pulumi code that operates on a JSON object and removes all of the policy statements by replacing it with an empty list.
+
+```javascript
+const jsonIAMPolicy = pulumi.output(`{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "s3:ListAllMyBuckets",
+                "s3:GetBucketLocation"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": "s3:*",
+            "Resource": "arn:aws:s3:::my-bucket"
+        }
+    ]
+}`);
+
+const policyWithNoStatements = pulumi.jsonParse(jsonIAMPolicy).apply(policy => {
+    // delete the policy statements
+    policy.Statement = [];
+    return policy;
+});
+```
+
+For more information view the NodeJS documentation.
+
+{{% /choosable %}}
+{{% choosable language typescript %}}
+
+### Example Converting Outputs to JSON
+
+The following example allows you to natively represent the definition of an AWS Step Function State Machine and easily embed outputs from other resources in your stack.
+
+```typescript
+const stateMachine = new awsnative.stepfunctions.StateMachine("stateMachine", {
+    roleArn: sfnRole.arn,
+    stateMachineType: "EXPRESS",
+    definitionString: pulumi.jsonStringify({
+        "Comment": "A Hello World example of the Amazon States Language using two AWS Lambda Functions",
+        "StartAt": "Hello",
+        "States": {
+            "Hello": {
+                "Type": "Task",
+                "Resource": helloFunction.arn,
+                "Next": "World",
+            },
+            "World": {
+                "Type": "Task",
+                "Resource": worldFunction.arn,
+                "End": true,
+            },
+        },
+    })
+});
+```
+
+### Example Parsing JSON Outputs to Objects
+
+The following is an example of Pulumi code that operates on a JSON object and removes all of the policy statements by replacing it with an empty list.
+
+```typescript
+const policyWithNoStatements = pulumi.jsonParse(jsonIAMPolicy).apply(policy => {
+    // delete the policy statements
+    policy.Statement = [];
+    return policy;
+});
+```
+
+For more information view the NodeJS documentation.
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+### Example Converting Outputs to JSON
+
+The following example allows you to natively represent the definition of an AWS Step Function State Machine and easily embed outputs from other resources in your stack.
```

Comment:
Repeating the header and the description between each chooser will become very hard to maintain.
Consider instead <header> <chooser with examples> <header> <chooser with examples>

```markdown
**Example converting Outputs to JSON**

{{< chooser ... >}}
...
{{< /chooser ... >}}

**Example parsing JSON Outputs to an object**

{{< chooser ... >}}
...
{{< /chooser ... >}}
```

At that point, if you wanted to go back to using headers, you could give them simpler/shorter names:

```markdown
### Serializing Outputs to JSON

{{< chooser ... >}}
...
{{< /chooser ... >}}

### Deserializing JSON Outputs

{{< chooser ... >}}
...
{{< /chooser ... >}}
```

Reply:
I think I can deduplicate it for the examples for `Deserializing JSON Outputs`.

However, `Serializing Outputs to JSON` has a different example in `Go` with a different description and an additional Note. I can deduplicate the Header, but maybe not the content.

@cnunciato Is there a solution to do the deduplication for this second case?

Reply:
Ah, I didn't realize the content varied for Go. Makes sense based on what you said while writing these docs.

Reply:
Nvm. I have a solution that I think looks good.